### PR TITLE
Make BlockSplitBloomFilter a view

### PR DESCRIPTION
**Pull request by PragmaTwice**
Besides, we supplement licenses and introduce span-lite to use std::span in c++20.

### CODE REVIEW

File: src/types/bloom_filter.h [hunk 1]
```diff
@@ -46,6 +49,40 @@ constexpr bool IsMultipleOf8(int64_t n) { return (n & 7) == 0; }
...
+/// Initialize the BlockSplitBloomFilter. The range of num_bytes should be within
+/// [kMinimumBloomFilterBytes, kMaximumBloomFilterBytes], it will be
+/// rounded up/down to lower/upper bound if num_bytes is out of range and also
+/// will be rounded up to a power of 2.
+///
+/// @param num_bytes The number of bytes to store Bloom filter bitset.
+OwnedBlockSplitBloomFilter CreateBlockSplitBloomFilter(uint32_t num_bytes);
+
+/// Initialize the BlockSplitBloomFilter. It copies the bitset as underlying
+/// bitset because the given bitset may not satisfy the 32-byte alignment requirement
```
Comment (mapleFU) [inline src/types/bloom_filter.h]:
```suggestion
/// bitset when the given bitset may not satisfy the 32-byte alignment requirement
```

Would this be better?

**Review by mapleFU — DISMISSED** [review]
+1

File: src/types/bloom_filter.h (hunk 1 above)
Comment (mapleFU) [reply src/types/bloom_filter.h]: Oh, seems this bloom never copy, and when copy it's happening, it doesn't have owned-`std::string` as return value?

Comment (PragmaTwice) [reply src/types/bloom_filter.h]: it is just a view, does not own anything. If you want to a owned filter, you can have a std::tuple<BloomFilter, std::string>.

File: src/types/bloom_filter.h [hunk 2]
```diff
@@ -46,6 +49,40 @@ constexpr bool IsMultipleOf8(int64_t n) { return (n & 7) == 0; }
...
+/// rounded up/down to lower/upper bound if num_bytes is out of range and also
+/// will be rounded up to a power of 2.
+///
+/// @param num_bytes The number of bytes to store Bloom filter bitset.
+OwnedBlockSplitBloomFilter CreateBlockSplitBloomFilter(uint32_t num_bytes);
+
+/// Initialize the BlockSplitBloomFilter. It copies the bitset as underlying
+/// bitset when the given bitset may not satisfy the 32-byte alignment requirement
+/// which may lead to segfault when performing SIMD instructions. It is the caller's
+/// responsibility to free the bitset passed in.
```
Comment (zncleon) [inline src/types/bloom_filter.h:70]: Maybe we should modify some comments like this?
```suggestion
/// responsibility to ensure the bitset would not be changed when using the bloomfilter.
```

Comment (PragmaTwice) [reply src/types/bloom_filter.h:70]: the bitset can be changed via `InsertHash`.

Comment (mapleFU) [reply src/types/bloom_filter.h:70]: I guess @zncleon means the mutability. Sometime we may need `const` ctor

Comment (PragmaTwice) [reply src/types/bloom_filter.h:70]: it seems not related to this function

Comment (mapleFU) [reply src/types/bloom_filter.h:70]: Yeah, I mean can we provide something like:

```
StatusOr<unique_ptr<BlockSplitBloomFilter>> CreateBlockSplitBloomFilter(uint8_t* bitset, uint32_t num_bytes);
StatusOr<unique_ptr<BlockSplitBloomFilter>> CreateBlockSplitBloomFilter(std::string& bitset);
StatusOr<unique_ptr<const BlockSplitBloomFilter>> CreateBlockSplitBloomFilter(const uint8_t* bitset, uint32_t num_bytes);
StatusOr<unique_ptr<const BlockSplitBloomFilter>> CreateBlockSplitBloomFilter(const std::string& bitset);
```

Comment (PragmaTwice) [reply src/types/bloom_filter.h:70]: if a readonly version is needed, you can have a `const BlockSplitBloomFilter`.

Comment (PragmaTwice) [reply src/types/bloom_filter.h:70]: Got your point. I think you can add it when you need it.

**Review by mapleFU — APPROVED** [review]

**Pull request merged**